### PR TITLE
Test speaker : catch exit code + shellcheck fixes

### PR DESCRIPTION
**Pull request by greg-intel**
Fixed a variety of quoting issues.

[Rewrite and reverse error conditions.](https://github.com/thesofproject/sof-test/commit/c10f5e9fef7461a8c1f32878809dea62a5eb53cb)

Signed off by Greg Galloway greg.galloway@intel.com

### CODE REVIEW

File: test-case/test-speaker.sh [hunk 1]
```diff
@@ -44,15 +44,13 @@ do
...
-    resultRet=$?
+    speaker-test -D "$dev" -r "$rate" -c "$channel" -f "$fmt" -l "$tcnt" -t wav -P 8 2>&1 |tee "$LOG_ROOT"/result_"$idx".txt
+    resultRet=${PIPESTATUS[0]}
 
-    if [[ $resultRet -eq 0 ]]; then
-        grep -nr -E "error|failed" $LOG_ROOT/result_$idx.txt
-        if [[ $? -eq 0 ]]; then
-            func_lib_lsof_error_dump $snd
+    if grep -nr -E "error|failed" "$LOG_ROOT"/result_"$idx".txt ||
+	[[ $resultRet -ne 0  ]]; then
```
Comment (marc-hb) [inline test-case/test-speaker.sh:51]: nit: indent this line

**Review by marc-hb — CHANGES_REQUESTED** [review]
Looks good except for the 2nd error message which is basically... missing. This is a bug fix! So it must mention the test name in the title and a short description of the bug, something like: "now checking exit status of speaker-test command.

File: test-case/test-speaker.sh (hunk 1 above)
Comment (greg-intel) [reply test-case/test-speaker.sh]: Fixed.

**Comment by greg-intel** [comment]
> Looks good except for the 2nd error message which is basically... missing. This is a bug fix! So it must mention the test name in the title and a short description of the bug, something like: "now checking exit status of speaker-test command"

The commit message?  Updated.

**Pull request merged**